hipercow-api/Controllers/ClustersController.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Imperial College London. All rights reserved.
+
+namespace Hipercow_api.Controllers
+{
+    using Hipercow_api.Models;
+    using Hipercow_api.Tools;
+    using Microsoft.AspNetCore.Mvc;
+
+    /// <summary>
+    /// The /clusters and /clusters/xxx endpoints provide the list of clusters
+    /// and information about a particular cluster respectively.
+    /// </summary>
+    [ApiController]
+    [Route("api/v1/[controller]")]
+    public class ClustersController : ControllerBase
+    {
+        /// <summary>
+        /// An object implementing IClusterInfoQuery, responsible for returning
+        /// the list of clusters, or information on a specific cluster.
+        /// </summary>
+        private IClusterInfoQuery clusterInfoQuery = new ClusterInfoQuery();
```

Comment:
should proabbly use dependency injection here... in program.cs `builder.Services.AddSingleton<IClusterInfoQuery, ClusterInfoQuery>();` .... and then here can inject into constructor.. also will allow you to mock and make testing easier
```
private IClusterInfoQuery clusterInfoQuery;

public ClustersController(IClusterInfoQuery clusterInfoQuery) 
{
this.clusterInfoQuery = clusterInfoQuery
}
```

Reply:
Think I have done this correctly in #3